networkit/cpp/planarity/LeftRightPlanarityTest.cpp

```diff
@@ -0,0 +1,261 @@
+//
+// Created by andreas on 03.01.25.
+//
+#include <networkit/planarity/LeftRightPlanarityTest.hpp>
+
+namespace NetworKit {
+
+void LeftRightPlanarityTest::run() {
+    // Euler-criterion
+    if (graph_->numberOfNodes() > 2 && graph_->numberOfEdges() > 3 * graph_->numberOfNodes() - 6) {
+        isPlanar_ = false;
+        return;
+    }
+
+    heights.assign(graph_->upperNodeIdBound(), noneHeight);
+    graph_->forNodes([&](const node currentNode) {
+        if (heights[currentNode] == noneHeight) {
+            heights[currentNode] = 0;
+            roots.push_back(currentNode);
+            this->dfsOrientation(currentNode);
+        }
+    });
+
+    sortAdjacencyListByNestingDepth();
+    isPlanar_ = true;
```

Comment:
Here you can use `std::ranges::all_of` https://en.cppreference.com/w/cpp/algorithm/ranges/all_any_none_of.